lxd/api_internal_recover.go

```diff
@@ -164,10 +164,11 @@ func internalRecoverScan(s *state.State, userPools []api.StoragePoolsPost, valid
 
 				// If pool doesn't exist in DB, initialise a temporary pool with the supplied info.
 				poolInfo := api.StoragePool{
-					Name:           p.Name,
-					Driver:         p.Driver,
-					StoragePoolPut: p.StoragePoolPut,
```

Comment:
We have a p.StoragePoolPut here so we should be using api.StoragePool.SetWritable() here to avoid duplication.